library/core/src/intrinsics.rs

```diff
@@ -2020,6 +2020,20 @@ extern "rust-intrinsic" {
     #[rustc_safe_intrinsic]
     pub fn saturating_sub<T: Copy>(a: T, b: T) -> T;
 
+    /// This is a *typed* read, `copy *p` in MIR.
+    ///
+    /// The stabilized form of this intrinsic is [`crate::ptr::read`], so
+    /// that can be implemented without needing to do an *untyped* copy
+    /// via [`copy_nonoverlapping`], and thus can get proper metadata.
+    ///
+    /// This intrinsic can *only* be called with a copy or move of a local.
+    /// (It allows neither constants nor projections.)
+    ///
+    /// To avoid introducing any `noalias` requirements, it just takes a pointer.
+    #[cfg(not(bootstrap))]
+    #[rustc_const_unstable(feature = "const_ptr_read", issue = "80377")]
+    pub fn read_via_copy<T>(p: *const T) -> T;
```

Comment:
Uhhh, is this callable as stable API right now?

Reply:
No, it's unstable, but without the `rustc_const_unstable` attribute the use in `ptr::read` fails with
```text
error[E0015]: cannot call non-const fn `intrinsics::read_via_copy::<T>` in constant functions
    --> library\core\src\ptr\mod.rs:1186:13
     |
1186 |             crate::intrinsics::read_via_copy(src)
     |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
     |
     = note: calls in constant functions are limited to constant functions, tuple structs and tuple variants
```

Reply:
Oh okay! I know that #[rustc_const_unstable]` adds const to an fn, I just wanted to check to make sure that we weren't adding stable callable API in the non-constant case.